lib/chef/knife/ec2_server_create.rb

```diff
@@ -739,130 +574,68 @@ def s3_validation_key
-        bootstrap.config[:first_boot_attributes_from_file] = locate_config_value(:first_boot_attributes_from_file)
-        bootstrap.config[:encrypted_data_bag_secret] = s3_secret || locate_config_value(:secret)
-        bootstrap.config[:encrypted_data_bag_secret_file] = locate_config_value(:secret_file)
+      def bootstrap_common_params
+        config[:encrypted_data_bag_secret] = s3_secret || config_value(:secret)
+        config[:encrypted_data_bag_secret_file] = config_value(:secret_file)
         # retrieving the secret from S3 is unique to knife-ec2, so we need to set "command line secret" to the value fetched from S3
         # When linux vm is spawned, the chef's secret option proc function sets the value "command line secret" and this value is used by
         # chef's code to check if secret option is passed through command line or not
-        Chef::Knife::DataBagSecretOptions.set_cl_secret(s3_secret) if locate_config_value(:s3_secret)
-        bootstrap.config[:secret] = s3_secret || locate_config_value(:secret)
-        bootstrap.config[:secret_file] = locate_config_value(:secret_file)
-        bootstrap.config[:node_ssl_verify_mode] = locate_config_value(:node_ssl_verify_mode)
-        bootstrap.config[:node_verify_api_cert] = locate_config_value(:node_verify_api_cert)
-        bootstrap.config[:bootstrap_no_proxy] = locate_config_value(:bootstrap_no_proxy)
-        bootstrap.config[:bootstrap_url] = locate_config_value(:bootstrap_url)
-        bootstrap.config[:bootstrap_install_command] = locate_config_value(:bootstrap_install_command)
-        bootstrap.config[:bootstrap_wget_options] = locate_config_value(:bootstrap_wget_options)
-        bootstrap.config[:bootstrap_curl_options] = locate_config_value(:bootstrap_curl_options)
-        bootstrap.config[:bootstrap_vault_file] = locate_config_value(:bootstrap_vault_file)
-        bootstrap.config[:bootstrap_vault_json] = locate_config_value(:bootstrap_vault_json)
-        bootstrap.config[:bootstrap_vault_item] = locate_config_value(:bootstrap_vault_item)
-        bootstrap.config[:bootstrap_template] = locate_config_value(:bootstrap_template)
-        bootstrap.config[:use_sudo_password] = locate_config_value(:use_sudo_password)
-        bootstrap.config[:yes] = locate_config_value(:yes)
+        Chef::Knife::DataBagSecretOptions.set_cl_secret(s3_secret) if config_value(:s3_secret)
+        config[:secret] = s3_secret || config_value(:secret)
 
         # If --chef-tag is provided then it will be set in chef as single value e.g. --chef-tag "myTag"
         # Otherwise if --tag-node-in-chef is provided then it will tag the chef in key=value pair of --tags option
         # e.g. --tags "key=value"
-        if locate_config_value(:chef_tag)
-          bootstrap.config[:tags] = locate_config_value(:chef_tag)
-        elsif locate_config_value(:tag_node_in_chef)
-          bootstrap.config[:tags] = config[:tags]
+        if config_value(:chef_tag)
```

Comment:
I think we should remove the option --tag-node-in-chef now https://github.com/chef/knife-ec2/pull/581/files#diff-e2993b8367db674f58a052036f00052aR792 as its already been deprecated. @btm what do you think ?

Reply:
@Vasu1105 @vsingh-msys yeah, we should drop the deprecated options. They've been deprecated over a year now. including `--security-group-ids`.